js/src/components/paid-ads/campaign-assets-form.js

```diff
@@ -90,13 +91,20 @@ export default function CampaignAssetsForm( {
 			baseAssetGroup,
 			validationRequestCount,
 			assetGroupErrors,
+			hasImportedAssets,
```

Comment:
I'm ok adding documentation. However, I find it self-descriptive `hasImportedAssets` clearly indicates (IMO at least) if yes/no assets were imported. 

Before adding the docs, I would love to clarify some points in case the logic behind my decisions is wrong or there are other more straightforward ways to achieve the same result. 

> Why it uses `assetGroup` rather than `nextAssetGroup` in the following `if` statement
 
This is because `nextAssetGroup` has a callback to `initialAssetGroup` that might have present assets on it, on the other hand, `assetGroup` are only the new assets imported. 

> Why it needs a separate state rather than hasImportedAssets: Boolean( baseAssetGroup[ ASSET_FORM_KEY.FINAL_URL ] ),

From my understanding, the presence of `ASSET_FORM_KEY.FINAL_URL` in `baseAssetGroup` only indicates that a new Final URL was loaded. So it would be lexically incorrect. Yes, It's true that right now in 100% of the cases, there will be imported assets since `BUSSINESS_NAME` is always set as `get_bloginfo( 'name' )`. However, I thought this might change in the future ( for example, if we won't import directly the Blog name)  then `ASSET_FORM_KEY.FINAL_URL` might be filled but might be that nothing was imported... 
 
 
 WDYT?

Reply:
> However, I find it self-descriptive `hasImportedAssets` clearly indicates (IMO at least) if yes/no assets were imported.

If without looking further into the design document and just inferring the business logic from the implementation, I would **guess** the rules for whether to display the `Tip` might be all of the following conditions are met:
1. There must be an action first that the user imports the suggested assets by selecting a Final URL via the UI.
2. At least one of the imported suggested assets is applied to the form's default values.
3. When editing assets that have been previously saved, assets got from Google Ads API are not considered as "using imported assets" because it doesn't meet the first rule: There must be an importing action first.

But these were only my **guess**, and I was still unsure what actual rules have to be fulfilled. IMO, the naming of `hasImportedAssets` alone doesn't clearly self-describe the above rules. The questions of "why" mentioned above were also questions I asked myself during my inference when the first round of code reviews.

If it seems to exist a totally equivalent state but was not used, there must be a particular reason why they were not used, otherwise, I would wonder if the developer missed a state that could be directly reused.

> > Why it needs a separate state rather than hasImportedAssets: Boolean( baseAssetGroup[ ASSET_FORM_KEY.FINAL_URL ] ),

> From my understanding, the presence of `ASSET_FORM_KEY.FINAL_URL` in `baseAssetGroup` only indicates that a new Final URL was loaded. So it would be lexically incorrect. [...] However, I thought this might change in the future ( for example, if we won't import directly the Blog name) then `ASSET_FORM_KEY.FINAL_URL` might be filled but might be that nothing was imported...

This is indeed different from my original guess, though they eventually can be achieved by the same implementation as the current one.

> WDYT?

From the design documents, I was/am not actually sure what the exact business logic is, but the current implementation and what the `Tip` is trying to convey seem reasonable.

Unless there were something missing, I think the current implementation is fine. Maybe just add the business logic as you understood it and why it checks all assets regardless of the backend implementation.

Reply:
Ok, I will add the mentioned explanations if you think they will help